job/service.go

```diff
@@ -694,6 +739,25 @@ func (srv *Service) fetchSpecsForGivenJobNames(ctx context.Context, projectSpec
 	return jobSpecs, nil
 }
 
+func (srv *Service) IsJobDestinationDuplicate(ctx context.Context, jobSpec models.JobSpec) (string, error) {
```

Comment:
ideally, function name with `Is` prefix indicates the function doing the check and returning a boolean value. let's rename this to something else, for example `GetJobNamesWithDuplicateDestination` or feel free to suggest if you have better name.

we also don't need to pass the whole JobSpec, `fullName` and `destination` should be enough.